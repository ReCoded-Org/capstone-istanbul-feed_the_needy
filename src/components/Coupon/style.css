.couponCard div {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.addBtn {
  background-color: #ff6600;
  border-color: #ff6600;
  --antd-wave-shadow-color: #ff6600;
}

.addBtn:hover,
.addBtn:focus {
  background-color: #ff5100;
  border-color: #ff6600;
}

.couponPrices {
  display: flex;
  align-self: center;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 32em;
  margin: 1em 1em;
}

.couponBody {
  flex-wrap: wrap;
  display: flex;
  justify-content: center;
}

.singlePrice {
  margin: 1em 1em;
}

.singlePrice:hover,
.singlePrice:focus {
  color: #262626;
  background: #fff8f0;
  border-color: #ff9125;
}

.ant-btn-primary.singlePrice {
  color: #fff;
  background: #ff6600;
  border-color: #ff6600;
}

.singlePrice > * {
  pointer-events: none;
}

.singleCouponModalButton {
  border: unset;
  background-color: unset;
}

.singleCouponModalButton:hover {
  cursor: pointer;
}
