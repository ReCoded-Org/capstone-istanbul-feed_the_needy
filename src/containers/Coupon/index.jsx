import React from "react"
import "antd/dist/antd.css"

const Coupon = () => {
  return <div>Coupon</div>
}

export default Coupon: