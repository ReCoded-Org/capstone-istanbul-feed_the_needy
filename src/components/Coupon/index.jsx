import React, { useState, useEffect } from "react";
import { Typography, Card, Button } from "antd";
import { useTranslation } from "react-i18next";
import "./style.css";

const { Title, Text } = Typography;

const Coupon = ({ id, createdAt, amounts, compName, img, isTesting }) => {
  const { t } = useTranslation();
  const [isSelected, setIsSelected] = useState({
    price0: false,
    price1: false,
    price2: false,
    price3: false,
  });
  const [selectedPrice, setSelectedPrice] = useState(0);
  const [cart, setCart] = useState([]);

  const handleSelect = (e) => {
    setIsSelected({ [e.target.name]: true });
    setSelectedPrice(parseInt(e.target.value, 10));
  };

  const handleAddToCart = () => {
    const cartCopy = [...cart];
    const isAdded = cartCopy.find(
      (cartCoupon) => cartCoupon.id === id + selectedPrice
    );
    if (!isAdded && selectedPrice) {
      const addedCoupn = {
        id: id + selectedPrice,
        createdAt,
        img,
        organizationName: compName,
        amount: selectedPrice,
      };
      cartCopy.push(addedCoupn);
      setCart(cartCopy);
      const stringCart = JSON.stringify(cartCopy);
      localStorage.setItem("cart", stringCart);
    }
  };

  useEffect(() => {
    const localCart = localStorage.getItem("cart");
    const parsedLocalCart = JSON.parse(localCart);
    if (parsedLocalCart) {
      setCart(parsedLocalCart);
    }
  }, []);

  return (
    <div className="coupon">
      <Title style={{ textAlign: "center" }} level={2}>
        {compName}
      </Title>
      <div className="couponBody">
        <Card className="couponCard" hoverable style={{ width: 350 }}>
          <img width={200} src={img} alt="logo" />
          <Text style={{ fontSize: "2em" }} strong>
            {selectedPrice
              ? selectedPrice + "₺"
              : t("couponsPage.coupon.amount")}
          </Text>
          {isTesting ? (
            <button type="button">Add to cart</button>
          ) : (
            <Button
              onClick={handleAddToCart}
              className="addBtn"
              size="large"
              block
              type="primary"
            >
              {t("couponsPage.coupon.addBtn")}
            </Button>
          )}
        </Card>
        <div className="couponPrices">
          {isTesting ? (
            <button type="button">price</button>
          ) : (
            <>
              {amounts.map((amount, index) => {
                return (
                  <Button
                    type={isSelected[`price${index}`] ? "primary" : "default"}
                    className="singlePrice"
                    size="large"
                    shape="round"
                    name={`price${index}`}
                    onClick={(e) => handleSelect(e)}
                    value={amount}
                  >
                    <b>{amount + "₺"}</b>
                  </Button>
                );
              })}
            </>
          )}
        </div>
      </div>
    </div>
  );
};

export default Coupon;
