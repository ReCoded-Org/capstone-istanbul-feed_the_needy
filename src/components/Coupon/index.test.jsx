import React from "react";
import renderer from "react-test-renderer";
import Coupon from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("Coupon renders correctly", () => {
  const tree = renderer.create(<Coupon isTesting />).toJSON();
  expect(tree).toMatchSnapshot();
});
