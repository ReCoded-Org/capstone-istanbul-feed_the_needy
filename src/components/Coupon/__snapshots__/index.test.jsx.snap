// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon renders correctly 1`] = `
<div
  className="coupon"
>
  <h2
    className="ant-typography"
    style={
      Object {
        "WebkitLineClamp": null,
        "textAlign": "center",
      }
    }
  >
    
  </h2>
  <div
    className="couponBody"
  >
    <div
      className="ant-card ant-card-bordered ant-card-hoverable couponCard"
      style={
        Object {
          "width": 350,
        }
      }
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <button
          className="singleCouponModalButton"
          onClick={[Function]}
          title="couponsPage.link"
          type="button"
        >
          <img
            alt="logo"
            width={200}
          />
        </button>
        <div
          className="singleCouponModal"
        />
        <span
          className="ant-typography"
          style={
            Object {
              "WebkitLineClamp": null,
              "fontSize": "2em",
            }
          }
        >
          <strong>
            couponsPage.coupon.amount
          </strong>
        </span>
        <button
          type="button"
        >
          Add to cart
        </button>
      </div>
    </div>
    <div
      className="couponPrices"
    >
      <button
        type="button"
      >
        price
      </button>
    </div>
  </div>
</div>
`;
